crates/net/network/src/transactions/mod.rs

```diff
@@ -1261,20 +1266,26 @@ where
             }
         }
 
+        let TxManagerPollDurations {
+            acc_network_events,
+            acc_pending_imports,
+            acc_tx_events,
+            acc_imported_txns,
+            acc_fetch_events,
+            acc_pending_fetch,
+            acc_cmds,
+        } = poll_durations;
+
         // update metrics for whole poll function
         this.metrics.duration_poll_tx_manager.set(start.elapsed());
         // update poll metrics for nested streams
-        this.metrics.acc_duration_poll_network_events.set(acc_network_events.as_micros() as f64);
-        this.metrics
-            .acc_duration_poll_pending_pool_imports
-            .set(acc_pending_imports.as_micros() as f64);
-        this.metrics.acc_duration_poll_transaction_events.set(acc_tx_events.as_micros() as f64);
-        this.metrics
-            .acc_duration_poll_imported_transactions
-            .set(acc_imported_txns.as_micros() as f64);
-        this.metrics.acc_duration_poll_fetch_events.set(acc_fetch_events.as_micros() as f64);
-        this.metrics.acc_duration_fetch_pending_hashes.set(acc_pending_fetch.as_micros() as f64);
-        this.metrics.acc_duration_poll_commands.set(acc_cmds.as_micros() as f64);
+        this.metrics.acc_duration_poll_network_events.set(acc_network_events.as_secs_f64());
+        this.metrics.acc_duration_poll_pending_pool_imports.set(acc_pending_imports.as_secs_f64());
+        this.metrics.acc_duration_poll_transaction_events.set(acc_tx_events.as_secs_f64());
+        this.metrics.acc_duration_poll_imported_transactions.set(acc_imported_txns.as_secs_f64());
+        this.metrics.acc_duration_poll_fetch_events.set(acc_fetch_events.as_secs_f64());
+        this.metrics.acc_duration_fetch_pending_hashes.set(acc_pending_fetch.as_secs_f64());
+        this.metrics.acc_duration_poll_commands.set(acc_cmds.as_secs_f64());
```

Comment:
if the duration metrics are also extracted into a struct, we could have a nice method like this:
```rust
fn update(&self, durations: TxManagerPollDurations) {
// ...
}
```
and it would be called like `this.metrics.duration_metrics.update(poll_durations)`